sdk/keyvault/azure-keyvault-administration/azure/keyvault/administration/_models.py

```diff
@@ -166,24 +166,39 @@ def __init__(self, **kwargs) -> None:
 
     @classmethod
     def _from_generated(
-        cls, response: "HttpResponse", deserialized_operation: "FullBackupOperation", response_headers: "Dict"
+        cls, response: HttpResponse, deserialized_operation: FullBackupOperation, response_headers: Dict
     ) -> "KeyVaultBackupResult":
         return cls(folder_url=deserialized_operation.azure_storage_blob_container_uri)
 
 
-class KeyVaultSetting(object):
+class KeyVaultSetting(Generic[T]):
     """A Key Vault setting.
 
     :ivar str name: The name of the account setting.
-    :ivar str value: The value of the pool setting.
-    :ivar SettingType type: The type specifier of the value.
+    :ivar value: The value of the pool setting. This is a boolean if `type` is SettingType.BOOLEAN, and a string
+        otherwise. If `value` is provided as a string when `type` is SettingType.BOOLEAN, the value will be converted
+        into a boolean (True if `value` is "True" (case-insensitive); False otherwise).
+    :vartype value: str or bool
+    :ivar type: The type specifier of the value.
+    :vartype type: SettingType or None
     """
 
-    def __init__(self, **kwargs) -> None:
-        self.name = kwargs.get("name")
-        self.value = kwargs.get("value")
-        self.type = kwargs.get("type")
+    def __init__(
+        self,
+        *,
+        name: str,
+        value: Union[str, bool],
+        type: Optional[SettingType] = None,
```

Comment:
Why would type be `Optional`? Don't we always know this?

Reply:
For the one vault setting that I've encountered so far, the `type` is always provided. The generated code unfortunately doesn't guarantee that it'll have a value, though: https://github.com/Azure/azure-sdk-for-python/blob/1148da2d5a75efa429bb7aaa002044d3b6180750/sdk/keyvault/azure-keyvault-administration/azure/keyvault/administration/_generated/v7_4_preview_1/models/_models_py3.py#L884-L886